pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -377,6 +390,8 @@ func (f *ring0Factory) BindFlags(flags *pflag.FlagSet) {
 	// to do that automatically for every subcommand.
 	flags.BoolVar(&f.clientCache.matchVersion, FlagMatchBinaryVersion, false, "Require server version to match client version")
 
+	f.BindDiscoveryFlags(flags)
```

Comment:
this should be `f.discoveryFactory.BindDiscoveryFlags(flags)`

from the interface doc: `each ring can depend on methods  an earlier ring, but cannot depend upon peer methods in its own ring`